CDP4Composition/Modularity/PluginUtilities.cs

```diff
@@ -102,5 +130,126 @@ public static DirectoryInfo PluginDirectoryExists(out bool specificPluginFolderE
 
             return directoryInfo;
         }
+
+        /// <summary>
+        /// Compute and return the <see cref="DirectoryInfo"/> of the temporary folder of the specified plugin
+        /// </summary>
+        /// <param name="pluginName">the name of the plugin</param>
+        /// <returns>a <see cref="DirectoryInfo"/></returns>
+        public static DirectoryInfo GetTempDirectoryInfo(string pluginName)
+        {
+            if (string.IsNullOrWhiteSpace(pluginName))
+            {
+                return null;
+            }
+
+            var temporaryFolder = Path.Combine(GetAppDataPath(), DownloadDirectory, "Temp");
+
+            if (!Directory.Exists(temporaryFolder))
+            {
+                Directory.CreateDirectory(temporaryFolder);
+            }
+
+            temporaryFolder = Path.Combine(temporaryFolder, pluginName);
+
+            return new DirectoryInfo(temporaryFolder);
+        }
+
+        /// <summary>
+        /// Retrieve all plugin that can be installed
+        /// </summary>
+        /// <returns>Returns a <see cref="IEnumerable{T}"/> of type <code>(FileInfo cdp4ckFile, Manifest manifest)</code> of the updatable plugins</returns>
+        public static IEnumerable<(FileInfo cdp4ckFile, Manifest manifest)> GetDownloadedInstallablePluginUpdate()
+        {
+            var downloadPath = Path.Combine(path1: GetAppDataPath(), path2: DownloadDirectory, PluginDirectoryName);
+
+            var updatablePlugins = new List<(FileInfo cdp4ckFile, Manifest manifest)>();
+
+            if (!Directory.Exists(path: downloadPath))
+            {
+                Logger.Info(message: "Download folder is empty or inexistant, download some plugins update from the IME first");
```

Comment:
Would you remove it then ?

Reply:
Well this message might indeed be obsolete. If you did not download any plugins yet, or maybe a user will never download it, then you get a lot of info messages in the logs. For debugging it could be interesting, but for the Info level I think it is definitely overkill.

Reply:
agreed